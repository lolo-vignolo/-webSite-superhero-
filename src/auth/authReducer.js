import React from 'react'
import { types } from '../types/types';

//estado si el usuario esta autenticado
// const state = {
//     name: "Lorenzo",
//     logged: true
// }


// const ejAction = {

//     type : "login",
//     payload : {
//         name : "newname",
//         logged : true or false
//     }
// }


export const authReducer = (state = {}, action ) => {
   

    switch (action.type) {
        case  types.login:
            
            return {
              ...action.payload,
              logged : true
          }
          
        case types.logout:
            
            return{
                logged:false
            }

        default:
        return state;



    }
}

