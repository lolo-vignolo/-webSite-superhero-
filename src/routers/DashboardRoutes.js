import React from 'react'
import { Routes, Route } from 'react-router-dom';
import { DcScreen } from '../components/dc/DcScreen'
import { HeroScreen } from '../components/heroes/HeroScreen'
import { MarvelScreen } from '../components/marvel/MarvelScreen'
import { SearchScreen } from '../components/search/SearchScreen'
import { Navbar } from '../components/ui/Navbar'

//Entre el sistema de rutas principal y uno secundarion, lo unico que la prncipal tiene un router


export const DashboardRoutes = () => {
    return (
        <>

<>
            <Navbar />

            <div className="container">
                <Routes>
                    <Route path="marvel" element={<MarvelScreen />} />
                    <Route path="dc" element={<DcScreen />} />

                    <Route path="search" element={<SearchScreen />} />
                    <Route path="hero/:heroeId" element={<HeroScreen />} />

                    <Route path="/" element={<MarvelScreen />} />

                </Routes>
            </div>
        </>
            
        </>
    )
}


{/* <Navbar />  
            <div className = "container-fluid mt-2 " >
                <Routes>
                    <Route path ="marvel" component={MarvelScreen} />
                    <Route path ="hero/:heroId" component={HeroScreen} />
                    <Route path ="dc" component={DcScreen} />
                    <Route path ="search" component={SearchScreen} />


                <Redirect to="/marvel" />
                </Routes>

            </div> */}
            