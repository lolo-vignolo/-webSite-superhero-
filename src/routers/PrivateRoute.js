import React, { useContext } from 'react';
import { Navigate, useLocation } from 'react-router';
import { AuthContext } from '../auth/AuthContext';






// ...rest , esto representa al path, exact y todas las caractreristicas que veo y uso dentro del <Route />.
const PrivateRoute = ({ children }) => {

    const { user } = useContext(AuthContext);
    const { pathname, search } = useLocation();
    
    localStorage.setItem('lastPath', pathname + search );
    
    return user.logged
        ? children
        : <Navigate to="/login" />
}


// el props representa a todos los props que he utilizado lo largo de la aplicacion como, history, location, params, search, etc.  
// componente con mayuscula representa al componente en si , ejemplo a AppRouter.js. 

export default PrivateRoute


// import { Redirect, Route } from 'react-router';
// import PropTypes from 'prop-types';

// const PrivateRoute =({
//     isAuthenticated,
//     component: Component,
//     ...rest
// }) => {

//     localStorage.setItem("lastPath", rest.location.pathname)
//     return(
//         <Route {...rest} 
//         component = { (props) => (

//             (isAuthenticated)
//                 ? (<Component {...props} />)
//                 : (<Redirect to = "/login" />)

//         ) }
//         />
 
//     )

// }


// PrivateRoute.propTypes = {
//     isAuthenticated: PropTypes.bool.isRequired,
//     component: PropTypes.func.isRequired
// }
