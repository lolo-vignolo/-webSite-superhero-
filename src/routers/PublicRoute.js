
import { useContext } from 'react';
import { Navigate } from 'react-router';

import { AuthContext } from '../auth/AuthContext';





export const PublicRoute = ({ children }) => {

    const { user } = useContext(AuthContext);

    
    return user.logged
        ? <Navigate to="/marvel" />
        : children
}







export default PublicRoute


// import React from 'react';
// import { Redirect, Route } from 'react-router';
// import PropTypes from 'prop-types';


// const PublicRoute = ({
//     isAuthenticated,
//     component: Component,
//     ...rest
// }) => {
//     return(
//         <Route {...rest} 
//         component = { (props) => (

//             (!isAuthenticated)
//                 ? (<Component {...props} />)
//                 : (<Redirect to = "/" />)


//         ) }
//         />


        
//     )
  
// }

// PublicRoute.propTypes = {
//     isAuthenticated: PropTypes.bool.isRequired,
//     component: PropTypes.func.isRequired
// }
