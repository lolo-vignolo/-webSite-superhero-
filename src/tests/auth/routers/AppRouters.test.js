import { mount} from "enzyme"
import { AuthContext } from "../../../auth/AuthContext";
import { AppRouter } from "../../../routers/AppRouter"


describe('test en mi <AppRouter />', () => {
    test('should shows login screeem if it is not auteticated', () => {
        
        const contextValue = {
            dispatch: jest.fn(),
            user:{
                logged: false
            }
            
        }

        const wrapper = mount (
        <AuthContext.Provider value = {contextValue} >
            <AppRouter />
        </AuthContext.Provider>
        );


        expect(wrapper).toMatchSnapshot();
        
    })

    test('should shows Marvel COmponent if it is autenticated', () => {
        const contextValue = {
            dispatch: jest.fn(),
            user:{
                logged: true,
                name : "Lorenzo"
            }
            
        }


        const wrapper = mount (
            <AuthContext.Provider value = {contextValue} >
                <AppRouter />
            </AuthContext.Provider>
            );
    
    
            expect(wrapper.find(".navbar").exists()).toBe(true);

        
    })
    
    
    
})
