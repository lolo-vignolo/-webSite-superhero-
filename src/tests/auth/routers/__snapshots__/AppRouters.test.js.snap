// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test en mi <AppRouter /> should shows login screeem if it is not auteticated 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      className="btn btn-secondary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
